File: simulator/src/model/Currency.ts
import Decimal from "decimal.js"

export default class Currency {

    private _name: string | null = null
    private _symbol: string | null = null
    private _lastPrice: Decimal | null = null
    private _lastPriceUpdateTime: Decimal | null = null

    constructor(instanceData?: Currency) {
        if (instanceData === undefined) return
        this._name = instanceData.name
        this._symbol = instanceData.symbol
        this._lastPrice = instanceData.lastPrice
        this._lastPriceUpdateTime = instanceData.lastPriceUpdateTime
    }

    public get name(): string | null {
        return this._name
    }
    public set name(value: string) {
        this._name = value
    }
    public get symbol(): string | null {
        return this._symbol
    }
    public set symbol(value: string) {
        this._symbol = value
    }
    public get lastPrice(): Decimal | null {
        return this._lastPrice
    }
    public set lastPrice(value: Decimal) {
        this._lastPrice = value
    }
    public get lastPriceUpdateTime(): Decimal | null {
        return this._lastPriceUpdateTime
    }
    public set lastPriceUpdateTime(value: Decimal) {
        this._lastPriceUpdateTime = value
    }
}

File: strategybuilder/src/main/java/com/gridy/strategybuilder/entity/Currency.java
package com.gridy.strategybuilder.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.util.Date;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Entity
@Table(name = "CURRENCY")
@EntityListeners(AuditingEntityListener.class)
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Currency {

  @Id
  @SequenceGenerator(name = "CURRENCY_ID_GENERATOR", sequenceName = "CURRENCY_ID_GEN", allocationSize = 1)
  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "CURRENCY_ID_GENERATOR")
  private Long id;

  @Column(name = "NAME", unique = true, nullable = false)
  private String name;

  @Column(name = "SYMBOL", unique = true, nullable = false)
  private String symbol;

  @Column(name = "LAST_PRICE")
  private BigDecimal lastPrice;

  @CreatedDate
  @Column(name = "CREATED_AT", nullable = false)
  private Date createdAt;

  @LastModifiedDate
  @Column(name = "UPDATED_AT", nullable = false)
  private Date updatedAt;
}
